scripts/make_sfc_climo.py

```diff
@@ -0,0 +1,107 @@
+"""
+The run script for SfcClimoGen
+"""
+
+import glob
+import os
+import sys
+from argparse import ArgumentParser
+from pathlib import Path
+
+from uwtools.api.fs import link as uwlink
+from uwtools.api.sfc_climo_gen import SfcClimoGen 
+from uwtools.api.config import get_yaml_config
+
+
+def link_files(dest_dir, files):
+    """
+    Link a given list of files to the destination directory using the same file names.
+    """
+    for fpath in files:
+        path = Path(fpath)
+        fn = Path(fpath).name
+    
+        if "halo" in fn:
+            fn = f"{CRES}.{(fn.replace('halo', 'halo4'))}"
+            no_halo_fn = fn.replace("halo4.", "")
+            for link in (fn, no_halo_fn):
+                link = Path(link)
+                if (linkname := dest_dir / link.name).is_symlink():
+                    linkname.unlink()
+                linkname.symlink_to(path)
+    
+        else:
+            bn = fn.split(".nc")[0]
+            fn = f"{CRES}.{bn}.halo0.nc"
+            tile1_fn = fn.replace("tile7.halo0", "tile1")
+            for link in (fn, tile1_fn):
+                link = Path(link)
+                if (linkname := dest_dir / link.name).is_symlink():
+                    linkname.unlink()
+                linkname.symlink_to(path)
+
+
+parser = ArgumentParser(
+    description="Script that runs SfcClimoGen via uwtools API.",
+)
+parser.add_argument(
+    "-c",
+    "--config-file",
+    metavar="PATH",
+    required=True,
+    help="Path to experiment config file.",
+    type=Path,
+)
+parser.add_argument(
+    "--key-path",
+    help="Dot-separated path of keys leading through the config to the driver's YAML block.",
+    metavar="KEY[.KEY...]",
+    required=True,
+    type=lambda s: s.split("."),
+)
+
+args = parser.parse_args()
+
+expt_config = get_yaml_config(args.config_file)
+CRES = expt_config["workflow"]["CRES"]
+os.environ["CRES"] = CRES 
+expt_config.dereference(
+    context={
+        **os.environ,
+        **expt_config,
+    }
+)
+
+args = parser.parse_args()
```

Comment:
I'm having a hard time wrapping this into its own function without getting an error message:
```
Failed with status: 126
sfc_climo_gen run: Final state: Not Ready
Will run sfc_climo_gen in {{ workflow.EXPTDIR/sfc_climo
```

Moving this block out and into the `make_sfc_climo` doesn't seem to work as expected. I've been playing around with it and comparing to the `mkae_orog` runscript. I'm thinking that the other functions need lines 65 - 73 to be available. Also wondering more about what this line from `make_orog` runscript does sepcifcally `args = parse_args(sys.argv[1:])`.

How the file is right now, it passes the fundamental test suite.